PythonVisualizations/Array.py

```diff
@@ -100,13 +111,24 @@ def insert(self, val):
         self.cleanUp(callEnviron)
 
     def removeFromEnd(self):
+        callEnviron = self.createCallEnvironment()
+        
+        #draw an index pointing to the last item in the list 
+        indexDisplay = self.createIndex(len(self.list)-1, 'nItems', level = -1, color = 'black')
+        callEnviron |= set(indexDisplay)         

```

Comment:
There's no need to recreate the `nItems` index in each operation.  Use the one built in the `display()` method.